model_server_tools/model_packaging/arg_parser.py

```diff
@@ -0,0 +1,105 @@
+# Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+"""
+This module parses the arguments given through the mxnet-model-server command-line. This is used by model-server
+at runtime.
+"""
+
+import argparse
+from model_server_tools.model_packaging.manifest_components.manifest import RuntimeType
+from model_server_tools.model_packaging.manifest_components.engine import EngineType
+
+
+class StoreDictKeyPair(argparse.Action):
+
+    """This class is a helper class to parse <model-name>=<model-uri> pairs
+    """
+
+    def __call__(self, parser, namespace, values, option_string=None):
+        try:
+            setattr(namespace, 'models', {kv.split('=', 1)[0]: kv.split('=', 1)[1] for kv in values})
+        except Exception:
+            raise Exception('Failed to parse <model=path>: ' + str(values) +
+                            'Format should be <model-name>=<model-path> (Local file path, URL, S3).')
+
+
+class ArgParser(object):
+
+    """Argument parser for model-export-tool commands
+    More detailed example is available at https://github.com/awslabs/mxnet-model-server/blob/master/README.md
+    """
+
+    @staticmethod
+    def export_model_args_parser():
+
+        """ Argument parser for mxnet-model-export
+        """
+        # TODO Add more CLI args here later
+        runtime_types = ', '.join(s.value for s in RuntimeType)
```

Comment:
This doesn't work for python 2.7

Reply:
Can you specify hat doesn't work here exactly for Python 2.7 ?
String.join ?

Reply:
It seems to work for me on both Py2 and Py 3.